include/af/image.h

```diff
@@ -688,6 +688,24 @@ AFAPI void moments(double* out, const array& in, const momentType moment=AF_MOME
 AFAPI array moments(const array& in, const momentType moment=AF_MOMENT_FIRST_ORDER);
 #endif
 
+#if AF_API_VERSION >= 35
+/**
+   C++ Interface for canny edge detector
+
+   \param[in] in is the input image
+   \param[in] lowThresholdRatio is the lower threshold % of maximum or auto-derived high threshold
+   \param[in] thresholdType determines if user set high threshold is to be used or not. It can take values defined by the enum \ref af_canny_threshold
+   \param[in] highThresholdRatio is the higher threshold % of maximum value in gradient image used in hysteresis procedure. This value is ignored if \ref AF_AUTO_OTSU_THRESHOLD is chosen as \ref af_canny_threshold
+   \param[in] sobelWindow is the window size of sobel kernel for computing gradient direction and magnitude
+   \param[in] isFast indicates if L<SUB>1</SUB> norm(faster but less accurate) is used to compute image gradient magnitude instead of L<SUB>2</SUB> norm.
+   \return binary array containing edges
+
+   \ingroup image_func_canny
+*/
+AFAPI array canny(const array& in, const float lowThresholdRatio,
+                  const CannyThreshold thresholdType, const float highThresholdRatio,
+                  const unsigned sobelWindow = 3, const bool isFast = false);
```

Comment:
As I said in the above comment, high threshold ratio will be either set manually by the user or computed using Otsu's algorithm - current options available via enum. 

Lower threshold ratio is a mandatory input from the user. I have put the threshold type enum input before high threshold parameter so that the user will notice the input value of threshold has an effect on the following high threshold ratio parameter.

If this is unsettling order, we can swap them. As long as we don't provide default value to threshold type parameter in C++ API, I don't have an objection with swapping the order of these variables. Because adding a default threshold type would confuse the user on how the algorithm behaves.